test/extended/registry/signature.go

```diff
@@ -0,0 +1,116 @@
+	defer g.GinkgoRecover()
+	var oc = exutil.NewCLI("registry-signing", exutil.KubeConfigPath())
+
+	g.It("can push a signed image to openshift registry and verify it", func() {
+		g.By("building an signer image that know how to sign images")
+		_, err := oc.Run("new-build").Args("--docker-image", "openshift/origin:latest", "--to", "signer:latest", "-D", "-").InputString(`FROM openshift/origin:latest
+ENV GNUPGHOME="/var/lib/origin/gnupg"
+RUN yum install -y skopeo && yum clean all
+RUN chmod 0777 -R /var/lib/origin && mkdir -p $GNUPGHOME && chmod 0600 $GNUPGHOME && \
+rm -f /dev/random; ln -sf /dev/urandom /dev/random && \
+echo $'%echo Generating a basic OpenPGP key ...\n\
+Key-Type: DSA \n\
+Key-Length: 1024 \n\
+Subkey-Type: ELG-E \n\
+Subkey-Length: 1024 \n\
+Name-Real: Joe Tester \n\
+Name-Comment: with stupid passphrase \n\
+Name-Email: joe@foo.bar \n\
+Expire-Date: 0 \n\
+Creation-Date: 2017-01-01 \n\
+%commit \n\
+%echo done \n'\
+>> $GNUPGHOME/setup_gpg && ls -lah /dev/random && ls -lah /dev/urandom && gpg2 --batch --gen-key $GNUPGHOME/setup_gpg && gpg2 --list-secret-keys`).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = exutil.WaitForAnImageStreamTag(oc, oc.Namespace(), "signer", "latest")
+		if err != nil {
+			containerLog, _ := oc.Run("logs").Args("builds/signer-1").Output()
+			e2e.Logf("signer build logs:\n%s\n", containerLog)
+		}
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("looking up the openshift registry URL")
+		registryURL, err := imagesutil.GetDockerRegistryURL(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("obtaining bearer token for the test user")
+		user := oc.Username()
+		token, err := oc.Run("whoami").Args("-t").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("granting the image-signer role to test user")
+		_, err = oc.AsAdmin().Run("adm").Args("policy", "add-cluster-role-to-user", "system:image-signer", oc.Username()).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("preparing the image stream where the signed image will be pushed")
+		_, err = oc.Run("create").Args("imagestream", "signed").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("signing the origin-pod:latest image and pushing it into signed:latest")
+		// TODO: Note that the skopeo command is expecting the KUBERNETES_MASTER and the
+		// KUBECONFIG set properly, so for that we need to `oc login` and we cannot use the
+		// service account (which is maybe good as we don't want to grant an service account
+		// image-signer and image-auditor roles.
+		signAndPushCmd := fmt.Sprintf(`export KUBERNETES_MASTER=https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT;`+
+			`oc login --token=%s $KUBERNETES_MASTER --certificate-authority=/run/secrets/kubernetes.io/serviceaccount/ca.crt && `+
+			`skopeo --debug --tls-verify=false copy --sign-by joe@foo.bar --dest-creds %s:%s --dest-tls-verify=false docker://docker.io/openshift/origin-pod:latest atomic:%s/%s/signed:latest`,
```

Comment:
Why is this signing `docker.io/openshift/origin-pod:latest`? Can we use some local image? Is this pulling the image or just a manifest?